src/north_mcp_python_sdk/auth.py

```diff
@@ -57,17 +58,24 @@ class AuthContextMiddleware:
     being stored in the context.
     """
 
-    def __init__(self, app: ASGIApp):
+    def __init__(self, app: ASGIApp, debug: bool = False):
         self.app = app
+        self.debug = debug
+        self.logger = logging.getLogger("NorthMCP.AuthContext")
+        if debug:
+            self.logger.setLevel(logging.DEBUG)
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send):
         if scope["type"] == "lifespan":
             return await self.app(scope, receive, send)
 
         user = scope.get("user")
         if not isinstance(user, AuthenticatedNorthUser):
+            self.logger.debug("Authentication failed: user not found in context. User type: %s", type(user))
```

Comment:
sorry for logger I meant you can do:
```python
self.logger.debug("Authentication failed: user not found in context", user_type=type(user))
```

this makes it possible for json/structured logs